service/src/test/java/cz/muni/fi/pa165/mushrooms/service/MushroomHunterServiceTest.java

```diff
@@ -0,0 +1,374 @@
+            };
+            minTimes = 0;
+
+
+            mushroomHunterDao.delete((MushroomHunter) any);
+            result = new Delegate() {
+                void foo(MushroomHunter mushroomHunter) {
+                    if (mushroomHunter == null || mushroomHunter.getId() == null ){
+                        throw new IllegalArgumentException("invalid entity");
+                    }
+                    if (persistedMushroomHunter.get(mushroomHunter.getId()) == null){
+                        throw new IllegalArgumentException("not in db");
+                    }
+                    persistedMushroomHunter.remove(mushroomHunter.getId());
+                }
+            };
+            minTimes = 0;
+
+            mushroomHunterDao.update((MushroomHunter) any);
+            result = new Delegate() {
+                void foo(MushroomHunter mushroomHunter) {
+                    if (mushroomHunter == null){
+                        throw new IllegalArgumentException("null mushroom hunter on mushroom hunter update");
+                    }
+                    if (mushroomHunter.getId() == null){
+                        throw new IllegalArgumentException("null id on mushroom hunter update");
+                    }
+                    if (mushroomHunter.getFirstName() == null){
+                        throw new IllegalArgumentException("null first name on mushroom hunter update");
+                    }
+                    if (mushroomHunter.getSurname() == null){
+                        throw new IllegalArgumentException("null surname on mushroom hunter update");
+                    }
+                    if (mushroomHunter.getUserNickname() == null){
+                        throw new IllegalArgumentException("null nickname on mushroom hunter update");
+                    }
+                    if (TestUtils.checkMushroomHunterDuplicity(persistedMushroomHunter, mushroomHunter)){
+                        throw new IllegalArgumentException("duplicating unique mushroom hunter attribute on updating");
+                    }
+                    persistedMushroomHunter.replace(mushroomHunter.getId(), mushroomHunter);
+                }
+            };
+            minTimes = 0;
+
+        }};
+
+        mushroomHunterDao.create(persistedHunter);
+        mushroomHunterDao.create(persistedHunter2);
+    }
+
+    @Test
+    public void findMushroomHunterById() throws Exception {
+        mushroomHunterDao.create(newHunter);
+        assertThat(service.findHunterById(newHunter.getId())).isEqualTo(newHunter);
+    }
+
+    @Test
+    public void findMushroomHunterByNullId() throws Exception {
+        assertThatThrownBy(() -> service.findHunterById(null)).isInstanceOf(IllegalArgumentException.class);
+    }
+
+    @Test
+    public void findMushroomHunterWithNonExistingId() throws Exception {
+        assertThat(service.findHunterById(-1L)).isNull();
+    }
+
+    @Test
+    public void findAllMushroomHunters() throws Exception {
```

Comment:
test not isolated - does create AND find -> change to only test find, or add such a test

Reply:
resolved